packages/lesswrong/components/users/DialogueSuggestionsPage.tsx

```diff
@@ -0,0 +1,782 @@
+  return userDialogueChecks?.find(check => check.targetUserId === targetUserId)?.checked || false;
+};
+
+type MatchDialogueButtonProps = {
+  isMatched: boolean;
+  targetUserId: string;
+  targetUserDisplayName: string;
+  currentUser: any; // replace with the correct type
+  loadingNewDialogue: boolean;
+  createDialogue: (title: string, participants: string[]) => void;
+  classes: ClassesType;
+};
+
+const MatchDialogueButton: React.FC<MatchDialogueButtonProps> = ({
+  isMatched,
+  targetUserId,
+  targetUserDisplayName,
+  currentUser,
+  loadingNewDialogue,
+  createDialogue,
+  classes,
+}) => {
+  return (
+    <div>
+      {isMatched ? (
+        <button
+          className={classes.newDialogueButton}
+          onClick={(e) =>
+            createDialogue(
+              `${currentUser?.displayName}/${targetUserDisplayName}`,
+              [targetUserId]
+            )
+          }
+        >
+          {loadingNewDialogue ? <a data-cy="message">Creating New Dialogue...</a> : <a data-cy="message">Start Dialogue</a>}
+        </button>
+      ) : null}
+    </div>
+  );
+};
+
+const MessageButton: React.FC<{
+  targetUserId: string;
+  currentUser: any; // replace with the correct type
+  classes: ClassesType;
+}> = ({ targetUserId, currentUser, classes }) => {
+  return (
+    <button className={classes.messageButton}>
+      <NewConversationButton user={{_id: targetUserId}} currentUser={currentUser}>
+        <a data-cy="message">Message</a>
+      </NewConversationButton>
+    </button>
+  );
+};
+
+export const DialogueSuggestionsPage = ({classes}: {
+  classes: ClassesType,
+}) => {
+  
+  const { captureEvent } = useTracking(); //it is virtuous to add analytics tracking to new components
+
+  const updateCurrentUser = useUpdateCurrentUser()
+  const [optIn, setOptIn] = React.useState(false); // for rendering the checkbox
+
+  const {create: createPost, loading: loadingNewDialogue, error: newDialogueError} = useCreate({ collectionName: "Posts", fragmentName: "PostsEdit" });
+  const { history } = useNavigation();
+
+  const { loading, error, data } = useQuery(gql`
+    query getDialogueUsers {
+      GetUserDialogueUsefulData {
+        dialogueUsers {
+          _id
+          displayName
+          karma
+        }
+        topUsers {
+          _id
+          displayName
+          total_power
+          total_agreement
+        }
+       }
+    }
+  `);
+
+  const userDialogueUsefulData : UserDialogueUsefulData = data?.GetUserDialogueUsefulData
+  const currentUser = useCurrentUser();
+
+  const {loading: userLoading, results : userDialogueChecks} = useMulti({
+    terms: {
+      view: "userDialogueChecks",
+      userId: currentUser?._id,
+      limit: 1000,
+    },
+    fragmentName: "DialogueCheckInfo",
+    collectionName: "DialogueChecks",
+  });
+
+  const [localDialogueChecks, setLocalDialogueChecks] = useState(userDialogueChecks ?? []);
```

Comment:
I think this was my test code, which may not be needed anymore.  (If you still need something like this to handle updating the state of the checkboxes on the client after submitting an update, that should probably be handled the same way you handle it on L312.)